src/nni_engine.hpp

```diff
@@ -68,33 +89,71 @@ class NNIEngine {
   void RunMainLoop();
   void RunPostLoop();
 
-  // ** Filter Functions
-
-  // Initialization step for filter before beginning
+  // ** Filtering Functions
+  // Initialization step for filter before beginning engine run.
   using StaticFilterInitFunction =
       std::function<void(NNIEngine &, GPEngine &, GraftDAG &)>;
   // Update step for any filter computation work to be done at start of each sweep.
   using StaticFilterUpdateFunction =
       std::function<void(NNIEngine &, GPEngine &, GraftDAG &)>;
-  // Function template for computational evaluation to be performed on an adjacent NNI.
-  using StaticFilterEvaluateFunction =
-      std::function<double(NNIEngine &, GPEngine &, GraftDAG &, const NNIOperation &)>;
   // Function template for processing an adjacent NNI to be accepted or rejected.
   using StaticFilterProcessFunction =
       std::function<bool(NNIEngine &, GPEngine &, GraftDAG &, const NNIOperation &)>;
 
-  // StaticFilterEvaluateFunction: dummy function that assigns all score of 0.0.
-  static double NoEvaluate(NNIEngine &this_nni_engine, GPEngine &this_gp_engine,
-                           GraftDAG &this_graft_dag, const NNIOperation &nni);
+  // StaticFilterInitFunction: Init function that does nothing.
+  static void NoInit(NNIEngine &this_nni_engine, GPEngine &this_gp_engine,
+                     GraftDAG &this_graft_dag) {}
+  // StaticFilterUpdateFunction: Update function that does nothing.
+  static void NoUpdate(NNIEngine &this_nni_engine, GPEngine &this_gp_engine,
+                       GraftDAG &this_graft_dag) {}
   // StaticFilterProcessFunction: dummy function that accepts all NNIs.
+  template <bool accept = true>
   static bool NoFilter(NNIEngine &this_nni_engine, GPEngine &this_gp_engine,
-                       GraftDAG &this_graft_dag, const NNIOperation &nni);
-  // StaticFilterEvaluateFunction: estimate an NNI's per GPCSP Likelihood using the
-  // Pre-NNI in the DAG.
-  static double EstimateNNILikelihoodViaPreNNI(NNIEngine &this_nni_engine,
-                                               GPEngine &this_gp_engine,
-                                               GraftDAG &this_graft_dag,
-                                               const NNIOperation &nni);
+                       GraftDAG &this_graft_dag, const NNIOperation &nni) {
+    return accept;
+  }
+
+  // Helper: Finds best pre-NNI likelihood in DAG for given NNI.
```

Comment:
To cpp, please

Reply:
Explain that the given nni could arise from multiple nnis from PCSPs already in the dag; this function gets the best one of those.

Reply:
Could mark for future consideration on the next round of PR work.

Reply:
Moved and added notes.